### PR TITLE
Upgrade smartcontract abi

### PR DESCRIPTION
**Pull request by kingpinXD**

### CODE REVIEW

File: chains/ethereum/contract/ETH/LockRedeem_Kratos.sol [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+
+    // internal functions
+    function addValidator(address v) internal {
+        validatorList.push(v);
+        validators[v] = uint8(validatorList.length);
+        emit AddValidator(v);
+    }
+
+    function MigrateFromOld() public {
+        require(msg.sender == old_contract);
```
Comment (ws4charlie) [inline chains/ethereum/contract/ETH/LockRedeem_Kratos.sol:257]: Is this expr to make sure address equality?

File: chains/ethereum/contract/ETH/LockRedeem_Kratos.sol [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+    //Approx
+    // 270 blocks per hour
+    // 5 blocks per min
+    // 28800 old value
+    uint256 LOCK_PERIOD;
+
+    // Variables for migrate
+    address old_contract;
+    uint signaturesrequiredformigration = 0;
+    uint migrationsignaturecount = 0;
```
Comment (ws4charlie) [inline chains/ethereum/contract/ETH/LockRedeem_Kratos.sol:45]: These 3 vars are extra params than existing script?

File: chains/ethereum/contract/ETH/LockRedeem_Kratos.sol (hunk 1 above)
Comment (kingpinXD) [reply chains/ethereum/contract/ETH/LockRedeem_Kratos.sol:257]: yes

File: chains/ethereum/contract/ETH/LockRedeem_Kratos.sol (hunk 2 above)
Comment (kingpinXD) [reply chains/ethereum/contract/ETH/LockRedeem_Kratos.sol:45]: Yes ,these are part of the the Kratos contract ,and not the Kainos one

File: chains/ethereum/contract/ETH/LockRedeem_Kratos.sol [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+    function addValidator(address v) internal {
+        validatorList.push(v);
+        validators[v] = uint8(validatorList.length);
+        emit AddValidator(v);
+    }
+
+    function MigrateFromOld() public {
+        require(msg.sender == old_contract);
+        migrationsignaturecount = migrationsignaturecount + 1;
+        //addValidator(tx.origin);
```
Comment (alexjlan) [inline chains/ethereum/contract/ETH/LockRedeem_Kratos.sol:259]: why this is removed?

Comment (alexjlan) [reply chains/ethereum/contract/ETH/LockRedeem_Kratos.sol:259]: after removing it, how can validator be added to the new contract?

Comment (kingpinXD) [reply chains/ethereum/contract/ETH/LockRedeem_Kratos.sol:259]: We are setting up new validators for kratos , so the new Validators will be added throiugh the constructor

**Pull request merged**